Packs/iDefense/Integrations/FeediDefense/FeediDefense.py

```diff
@@ -0,0 +1,161 @@
+
+GLOBAL_INTEGRATION_CONTEXT: dict = {}  # Global variable in order to hold one dictionary that manage all indicator types
+
+
+def _get_fetch_time() -> int:
+    try:
+        fetch_time = int(demisto.params().get('fetch_time', ''))
+    except ValueError:
+        fetch_time = 14
+    return fetch_time
+
+
+def _update_global_integration_context(key: str = None, value: str = None) -> None:
+    global GLOBAL_INTEGRATION_CONTEXT
+    if not key and not value:
+        GLOBAL_INTEGRATION_CONTEXT = get_integration_context()
+    else:
+        GLOBAL_INTEGRATION_CONTEXT[key] = value
+
+
+def _get_global_integration_context_by_key(key: str) -> Optional[Any]:
+    return GLOBAL_INTEGRATION_CONTEXT.get(key)
+
+
+def _get_global_integration_context() -> dict:
+    return GLOBAL_INTEGRATION_CONTEXT
+
+
+def custom_build_iterator(client: Client, feed: Dict, limit, **kwargs) -> List:
+    """
+    Implement the http_request with api that works with pagination and filtering. Uses GLOBAL_INTEGRATION_CONTEXT to
+    save last fetch time to each indicator type
+    Args:
+        client: Client manage all http requests
+        feed: dictionary holds all data needed to the specific service (Services- IP, Domain, URL)
+        limit: maximum number of indicators to fetch
+
+    Returns:
+        list of indicators returned from api. Each indicator is represented in dictionary
+    """
+    current_datetime = datetime.now()
+    fetch_time = _get_fetch_time()
+    params: dict = feed.get('filters', {})
+    current_indicator_type = feed.get('indicator_type', '')
+    _update_global_integration_context()
+    last_fetch = _get_global_integration_context_by_key(f'{current_indicator_type}_fetch_time')
+    page_number = 1
+    params['end_date'] = current_datetime.isoformat() + 'Z'
+    params['start_date'] = last_fetch if last_fetch else \
+        (current_datetime - timedelta(days=fetch_time)).isoformat() + 'Z'
+    params['page_size'] = 200
+
+    if not limit:
+        limit = 10000
+        _update_global_integration_context(current_indicator_type + '_fetch_time', str(params['end_date']))
+        set_integration_context(_get_global_integration_context())
+
+    more_indicators = True
+    result: list = []
+
+    while more_indicators:
+        params['page'] = page_number
+        r = requests.get(
+            url=feed.get('url', client.url),
+            verify=client.verify,
+            auth=client.auth,
+            cert=client.cert,
+            headers=client.headers,
+            params=params,
+            **kwargs
+        )
+
+        demisto.debug(f"initiating API call with url: {r.url}")
+        try:
+            r.raise_for_status()
+            data = r.json()
+            if data.get('total_size'):
+                result.extend(jmespath.search(expression=feed.get('extractor'), data=data))
+            more_indicators = data.get('more')
+            page_number += 1
+            if len(result) >= limit:
+                break
```

Comment:
don't you need to update the integration context with the page number? let's sync on that.